PDX-283: NUTs have been restructured as It was failing after repos restructuring

priyaprovar opened the pull request:
RCA: NUTs have been restructured as It was failing after repos restructuring
Fix: Removed the dependencies of generate , load, and set command. For this I have added a boilerplate json file, written load value in config by NUTs and Updated value in boilerplate json using NUTs

Palak-Bansal-Provar commented on test/commands/provar/automation/metadata/download1.nut.ts:
I think we can delete this file.

priyaprovar replied:
Yes , we can delete it as it is not being used anywhere.

priyaprovar replied:
deleted download1.nut.ts

Palak-Bansal-Provar commented on provardx-properties.json:
If we are using the boilerplate JSON here only that is with no custom set attributes then, we use constants file propertyFileContent.ts from the utils package, import, and use it.
As, If in future we change anything in the boilerplate JSON which is not good, it won't fail these test cases , ideally it should, so we should use that only.

priyaprovar replied:
We need  boilerplate JSON file location, so removed the content from json, left it blank and assigned propertyFileContent into this.

priyaprovar replied:
Removed boilerplate JSON file from project location and generated through NUT

Palak-Bansal-Provar commented on test/commands/provar/automation/project/compile.nut.ts:
```diff
@@ -55,22 +70,33 @@ describe('provar automation project compile NUTs', () => {
+    interface PropertyFileJsonData {
+      [key: string]: string | boolean | number;
+    }
+    const jsonFilePath = FILE_PATHS.PROVARDX_PROPERTIES_FILE;
```
if this variable is needed at multiple places, we can have it as a class variable

priyaprovar replied:
Done

Palak-Bansal-Provar commented on test/commands/provar/automation/metadata/download.nut.ts:
```diff
@@ -1,194 +1,332 @@
-        }
-      });
-    });
+  it('Update Provar Home and ProjectPath in provar dx properties file', async () => {
```
it() is something that executes a scenario that is it is a single NUT in itself. So, If we are not executing any scenario here, and just setting up the pre-requisites for further NUTs, then we should have all this in before() method

priyaprovar replied:
Added in before method

Palak-Bansal-Provar commented on test/commands/provar/automation/metadata/download.nut.ts:
```diff
@@ -1,194 +1,332 @@
-      });
-    });
+  it('Update Provar Home and ProjectPath in provar dx properties file', async () => {
+    const ProvarDXPropertiesFilePath1 = path.join(process.cwd(), './provardx-properties.json');
```
we are using camelCasing for variables like provarDXPropertiesFilePath.
please Check this for other places as well.

priyaprovar replied:
Changed for all variable which was not following the camelCasing rule

Palak-Bansal-Provar commented on test/commands/provar/automation/metadata/download.nut.ts:
```diff
@@ -1,194 +1,332 @@
+    await UpdateFileConfigSfdx();
+    async function UpdateFileConfigSfdx(): Promise<void> {
+      try {
+        // Read the directory and find the config.json file
```
I think we can remove all these comments at all places as the code is self-describing.

priyaprovar replied:
Removed comment from all files

Palak-Bansal-Provar commented on test/commands/provar/automation/metadata/download.nut.ts:
```diff
@@ -1,194 +1,332 @@
+  });
+
+  it('Metadata should not be downloaded as provarHome & projectPath are not correct and return the error message', async () => {
+    const ProvarDXPropertiesFilePath = path.join(process.cwd(), './provardx-properties.json');
+
+    // Read the config.json file
+    const ProvarDXPropertiesFileData = await fileSystem.readFile(ProvarDXPropertiesFilePath, { encoding: 'utf8' });
+
+    // Parse the JSON data
+    const ProvarDXPropertiesFileParsed = JSON.parse(ProvarDXPropertiesFileData);
+    ProvarDXPropertiesFileParsed['provarHome'] = './ProvarHome1';
+
+    // Convert the updated JSON object back to a string
+    const updatedPropertiesFileData = JSON.stringify(ProvarDXPropertiesFileParsed, null, 4);
+
+    // Write the updated JSON string back to the config.json file
+    await fileSystem.writeFile(ProvarDXPropertiesFilePath, updatedPropertiesFileData, 'utf8');
```
I think it is there in every NUT
If we are doing these operations at multiple places that is at multiple NUTs , either we can have it in before() method if used in mostly all NUTs or can derive a method and call it at every place.

priyaprovar replied:
It is not in every NUT and also wherever this type of line present , this is updating the boilerplate value for different key for negative scenario checking.

Palak-Bansal-Provar commented on test/commands/provar/automation/metadata/download.nut.ts:
```diff
@@ -1,194 +1,332 @@
 
-  it('Metadata should not be downloaded for the invalid user name and return the error', () => {
+  it('It should Delete the PROVARDX_PROPERTIES_FILE_PATH in sf config.json', async () => {
+    async function deleteFileConfigSfdx(): Promise<void> {
```
we have these types of functions as inner functions in multiple NUTs. these seem to be generic functions that can be also used further, so keep these functions outside it().
Make them a function in a separate class instead of creating it as an inner function.

priyaprovar replied:
This looks like generic function but this need to used only once so added in before

Palak-Bansal-Provar commented on test/commands/provar/automation/test/run.nut.ts:
```diff
@@ -1,50 +1,169 @@
-        }
-      });
+  it('Boilerplate json file should not run if the file has not been loaded', () => {
+    // Read the config.json file
```
remove the comments

priyaprovar replied:
removed comments from all files

Palak-Bansal-Provar reviewed (commented):
Please check for failing pipeline before merging

priyaprovar commented:
RCA: NUTs have been restructured as It was failing after repos restructuring
Fix: Removed the dependencies of generate , load, and set command. For this I have added a boilerplate json file, written load value in config by NUTs and Updated value in boilerplate json using NUTs
